src/platform/ESP32/NetworkCommissioningDriver.cpp

```diff
@@ -236,6 +236,17 @@ CHIP_ERROR ESPWiFiDriver::ConnectWiFiNetwork(const char * ssid, uint8_t ssidLen,
     return ConnectivityMgr().SetWiFiStationMode(ConnectivityManager::kWiFiStationMode_Enabled);
 }
 
+CHIP_ERROR ESPWiFiDriver::DisconnectDriver()
```

Comment:
This should be called `DisconnectFromNetwork`

Reply:
There must be a public interface change for DisconnectFromNetwork() to be an abstract base class method, overridden by platforms. We can't have a new feature that only is implemented/implementable on a single platform.

Reply:
Agree, though I think we can have a base class impl that just returns CHIP_ERROR_NOT_IMPLEMENTED since these code paths won't be relevant for single-interface devices.

Reply:
Added this change to https://github.com/project-chip/connectedhomeip/pull/35256, will close this later.